### PR TITLE
SCRIPTS Added The Missing Piece Quest (Teleport-Altep)

### PR DESCRIPTION
**Pull request by SSJ4Puar**
Had to edit a few NPCs in rabao and Norhern San d'Oria. Also has to create a new lua file for qm5 in quicksand caves. Tested it with an account on my local server and was unable to break it.

### CODE REVIEW

**Comment by xdemolish** [comment]
squash > amend last commit > change commit message to include SCRIPTS label e.g. ```SCRIPTS: added The Missing Piece (Teleport Altep) quest``` > force push > drop a comment so we know to take a look

File: scripts/zones/Northern_San_dOria/npcs/Charlaimagnat.lua [hunk 1]
```diff
@@ -32,7 +33,21 @@ end;
...
 
 function onTrigger(player,npc)
-	player:startEvent(0x02be);
+
+	local realday = tonumber(tostring(os.date("%Y")) .. os.date("%m") .. os.date("%d"));
+	local TheMissingPiece = player:getQuestStatus(OUTLANDS,THE_MISSING_PIECE);
+	
+	if(TheMissingPiece == QUEST_ACCEPTED and player:hasKeyItem(TABLET_OF_ANCIENT_MAGIC) and player:hasKeyItem(LETTER_FROM_ALFESAR)) then
+		player:startEvent(0x02bf);--Continuing the Quest
+		player:setVar("TheMissingPiece_date", 0);
```
Comment (whasf) [inline scripts/zones/Northern_San_dOria/npcs/Charlaimagnat.lua]: Why do you set it to zero here, but in onEventFinish you set it again? I'd just delete this line

Comment (SSJ4Puar) [inline scripts/zones/Northern_San_dOria/npcs/Charlaimagnat.lua]: i suppose it was left behind when i was coding the quest to start. i was just trying to figure out how setVar worked. Is there a way to delete vars after they've been used? I dont think anyone will be repeating this quest after its completed.

Comment (TeoTwawki) [inline scripts/zones/Northern_San_dOria/npcs/Charlaimagnat.lua]: > Is there a way to delete vars after they've been used?

Anytime you setVar() to zero it deletes it form the database.

**Comment by teschnei** [comment]
the scripts look good, just waiting on you to squash your commits and fix the commit name

**Comment by SSJ4Puar** [comment]
Is there a way to use the web interface to squash the commits or do i have to do that via Gitshell? I was able to get the files to my repo fairly easily but I'm not sure how to squash the commits on Github from the shell for you guys.

**Comment by TeoTwawki** [comment]
No there is not. Both the web interface and the github for windows program are utter crap.Git bash (command line, part of git for windows installation) or tortoise git (windows right click menu) can do it.

**Comment by TeoTwawki** [comment]
https://msysgit.github.io/
https://code.google.com/p/tortoisegit/

**Comment by SSJ4Puar** [comment]
Ok i'll have to look at it when im at home and have some free time. may not get a change to look at it until tuesday tho. we'll see. started working on the Fukuro quest as well but ill finish up this quest and get it submitted before i attempt to commit that one. 

Thanks for the help.

**Comment by TeoTwawki** [comment]
Tip: If you never commit anything to master and only use it to pull in updates from DSP's master you can instead always create new branches for whatever changes you make for pull requests, and an extra branch for your server if you need to commit any customizations. When you create a new branch using githubs website interface it will always start off as a 1:1 copy of your master branch (not dsp's master branch) so any changes you have in master will wind up replicated to the new branches. This makes it easy to have multiple branches to have multiple separate pull requests up at once without interfering with each other.

**Comment by SSJ4Puar** [comment]
I think i did it on using tortoisegit. Did you guys get a request? does it look ok or is there something i did wrong? :(

**Comment by TeoTwawki** [comment]
Looks like you squashed correctly.

**Comment by SSJ4Puar** [comment]
ill have to figure all this stuff out on tortoisegit. always been better with guis for this stuff

**Pull request merged**